services/motion/builtin/builtin_test.go

```diff
@@ -1400,6 +1401,170 @@ func TestMoveOnGlobe(t *testing.T) {
 	})
 }
 
+func TestMoveOnMapStaticObs(t *testing.T) {
+	ctx := context.Background()
+	logger := logging.NewTestLogger(t)
+	extra := map[string]interface{}{
+		"motion_profile": "position_only",
+		"timeout":        15.,
+		"smooth_iter":    10.,
+	}
+
+	baseName := "test-base"
+	slamName := "test-slam"
+
+	// Create an injected Base
+	geometry, err := (&spatialmath.GeometryConfig{R: 30}).ParseConfig()
+	test.That(t, err, test.ShouldBeNil)
+
+	injectBase := inject.NewBase(baseName)
+	injectBase.GeometriesFunc = func(ctx context.Context) ([]spatialmath.Geometry, error) {
+		return []spatialmath.Geometry{geometry}, nil
+	}
+	injectBase.PropertiesFunc = func(ctx context.Context, extra map[string]interface{}) (base.Properties, error) {
+		return base.Properties{TurningRadiusMeters: 0, WidthMeters: 0.6}, nil
+	}
+
+	// Create a base link
+	baseLink := createBaseLink(t)
+
+	// Create an injected SLAM
+	injectSlam := createInjectedSlam(slamName, "pointcloud/octagonspace.pcd", nil)
+	injectSlam.PositionFunc = func(ctx context.Context) (spatialmath.Pose, string, error) {
+		return spatialmath.NewPose(
+			r3.Vector{X: 0.58772e3, Y: -0.80826e3, Z: 0},
+			&spatialmath.OrientationVectorDegrees{OZ: 1, Theta: 90},
+		), "", nil
+	}
+
+	// Create a motion service
+	deps := resource.Dependencies{injectBase.Name(): injectBase, injectSlam.Name(): injectSlam}
+	fsParts := []*referenceframe.FrameSystemPart{{FrameConfig: baseLink}}
+
+	ms, err := NewBuiltIn(ctx, deps, resource.Config{ConvertedAttributes: &Config{}}, logger)
+	test.That(t, err, test.ShouldBeNil)
+	defer ms.Close(context.Background())
+
+	fsSvc, err := createFrameSystemService(ctx, deps, fsParts, logger)
+	test.That(t, err, test.ShouldBeNil)
+	ms.(*builtIn).fsService = fsSvc
+
+	goal := spatialmath.NewPoseFromPoint(r3.Vector{X: 0.6556e3, Y: 0.64152e3})
+
+	req := motion.MoveOnMapReq{
+		ComponentName: injectBase.Name(),
+		Destination:   goal,
+		SlamName:      injectSlam.Name(),
+		MotionCfg:     &motion.MotionConfiguration{PlanDeviationMM: 0.01},
+		Extra:         extra,
+	}
+
+	t.Run("one obstacle", func(t *testing.T) {
```

Comment:
Adding context for the following test case
the robot would be in the following octagon
![og](https://github.com/viamrobotics/rdk/assets/46872531/f59f1775-c4f3-4425-9ba9-a379721fdc4e)

I would then define a static obstacle which would generate a path which curves to the right
![1](https://github.com/viamrobotics/rdk/assets/46872531/7aabad1b-e52b-45c4-b8d5-7361cdb6375b)

i would then use the CheckPlan function to show that with the following new set of obstacles the plan fails proving that the path went to the right and avoided the obstacle on the left which we defined at run time
![wts fail](https://github.com/viamrobotics/rdk/assets/46872531/b9bbd42d-37a5-45e1-a6c7-74a23ce954c1)